addons/web/static/src/js/core/popover.js

```diff
@@ -70,7 +117,14 @@ odoo.define('web.Popover', function () {
                 .find((pos) => {
                     this.popoverRef.el.style.top = `${pos.top}px`;
                     this.popoverRef.el.style.left = `${pos.left}px`;
-                    return this.constructor._isInViewport(this.popoverRef.el);
+                    const rect = this.popoverRef.el.getBoundingClientRect();
```

Comment:
I though you needed to be able to override this in tests. A static function _isInViewPort that can be overridden to always return true in tests was the easiest way to achieve what you wanted, imho. But I may have misunderstood the needs.

Reply:
I don't need to override it in the tests at the end, because when it matters, the test is positioning the element in the viewport manually instead. Basically I just reverted to the old code of the tests since you guys weren't too excited with the env change, I thought it would be an improvement but I just discarded it.

The only difference is that previously the tests that didn't care about the positioning also needed the element to be in viewport to show the popover, and now they don't care. Instead of complaining that the target is not in viewport, the component is just picking the default position instead. That change also fixed a bug where during some resize situations, the popover would briefly go outside of viewport and disappear even though it should eventually show again, which is does now.